### PR TITLE
Antibody updates:

### LINKED ISSUE #357: db: update the antibody table fields to the new DWG specification
see: 

LINCS DWG Metadata Standards templates: [Antibody sheet](https://docs.google.com/spreadsheets/d/1pGZyOSYEf81MIvs9ve4jZLWgE7zCXPQ0IC3vkOTkqaQ/edit#gid=1842017586)
and

HMS LINCS [Antibody Specification sheet](https://docs.google.com/spreadsheets/d/16oVSgfcodNlK8iMKcoaIjtallRypws4JM-PqJVaMjHA/edit#gid=1012548304)

### PR DESCRIPTION
**Pull request by seanderickson**
* field meta information updates
* antibody model properties for target protein information
* antibody target protein link fields
* change date_type 'otherreagent' to 'other_reagent'
* resolves #357

### CODE REVIEW

File: django/db/models.py [hunk 1]
```diff
@@ -565,6 +565,26 @@ class Antibody(Reagent):
     labeling_details        = _TEXT(**_NULLOKSTR) 
     relevant_citations      = _TEXT(**_NULLOKSTR) 
     
+    @property
```
Comment (jmuhlich) [inline django/db/models.py]: It's unclear whether you intended to use the default None return value when these if's fall through, or you just forgot to write the else case. I am pretty sure you intended them to fall through and return None, so I would like to see an explicit `else` with `return None` on each of these. The other acceptable option is a ternary if/else expression, i.e. `return self.target_protein.name if self.target_protein else None`.

File: django/db/views.py [hunk 2]
```diff
@@ -2328,27 +2333,29 @@ def export_as_xls(name,col_key_name_map, cursor=None, queryset=None,
...
+                    temp.append(_write_val_safe(column))
+            ws.append(temp)
             if(row % debug_interval == 0):
                 logger.info("row: " + str(row))
             row += 1    
-    
-    wbk.save(response)
-    return response
 
+    response = HttpResponse(save_virtual_workbook(wb), content_type='application/vnd.ms-excel')
```
Comment (jmuhlich) [inline django/db/views.py]: That's not the right mime type for xlsx. It should be `application/vnd.openxmlformats-officedocument.spreadsheetml.sheet` according to https://technet.microsoft.com/en-us/library/ee309278%28office.12%29.aspx .

File: django/db/views.py [hunk 3]
```diff
@@ -2328,27 +2333,29 @@ def export_as_xls(name,col_key_name_map, cursor=None, queryset=None,
...
+    response['Content-Disposition'] = 'attachment; filename=%s.xlsx' % unicode(name).replace('.', '_')
```
Comment (jmuhlich) [inline django/db/views.py]: Why the conversion to `unicode` here? Where does `name` come from? It should either be a unicode object (in which case you want to call `name.encode('utf-8')`) or a str object already encoded in utf-8 in which case you should do nothing.

Comment (seanderickson) [inline django/db/views.py]: I'd like to remove non-word characters from the filename, and replace them with underscores; so I guess a better method would be:
`response['Content-Disposition'] = 'attachment; filename=%s.xlsx' % re.sub(r'\W','_',name)`

Comment (jmuhlich) [inline django/db/views.py]: Oops I misspoke. Only ASCII is reliably supported in HTTP headers, so any non-ASCII actually needs to be stripped. The question remains, where does `name` come from? It seems like it's composed from strings that are totally under our control and should never end up with non-ASCII characters in it in the first place.

Comment (jmuhlich) [inline django/db/views.py]: Yes, the `re.sub` approach makes more sense.

Comment (seanderickson) [inline django/db/views.py]: yes - it is totally under our control - and should always be composed of ascii chars (based on the code in place), so this is meant as a catch all/final check. Ideally, a constraint (assertion?) would be imposed - ?

**Comment by jmuhlich** [comment]
:+1: fire away!

**Pull request merged**